app/models/user/User.scala

```diff
@@ -41,29 +35,111 @@ case class UserSQL(
                   isDeactivated: Boolean,
                   created: Long = System.currentTimeMillis(),
                   isDeleted: Boolean = false
-                  )
+                  ) extends DBAccessContextPayload with Identity with FoxImplicits {
+  val name = firstName + " " + lastName
+
+  val abreviatedName =
+    (firstName.take(1) + lastName).toLowerCase.replace(" ", "_")
+
+  def organization = OrganizationSQLDAO.findOne(_organization)(GlobalAccessContext)
+
+  def experiences = UserExperiencesSQLDAO.findAllExperiencesForUser(_id)(GlobalAccessContext)
+
+  def userConfigurationStructured =
+    JsonHelper.jsResultToFox(userConfiguration.validate[Map[String, JsValue]]).map(UserConfiguration(_))
+
+  def teamMemberships = UserTeamRolesSQLDAO.findTeamMembershipsForUser(_id)(GlobalAccessContext)
+
+  def teamManagerMemberships =
+    for {
+      teamMemberships <- teamMemberships
+    } yield teamMemberships.filter(_.isTeamManager)
+
+  def teamManagerTeamIds =
+    for {
+      teamManagerMemberships <- teamManagerMemberships
+    } yield teamManagerMemberships.map(_.teamId)
+
+  def teamIds =
+    for {
+      teamMemberships <- teamMemberships
+    } yield teamMemberships.map(_.teamId)
+
+  def isTeamManagerOrAdminOf(otherUser: UserSQL): Fox[Boolean] =
+    for {
+      otherUserTeamIds <- otherUser.teamIds
+      teamManagerTeamIds <- teamManagerTeamIds
+    } yield (otherUserTeamIds.intersect(teamManagerTeamIds).nonEmpty || this.isAdminOf(otherUser))
 
-object UserSQL {
-  def fromUser(user: User)(implicit ctx: DBAccessContext): Fox[UserSQL] =
+  def isTeamManagerOrAdminOf(_team: ObjectId): Fox[Boolean] =
     for {
-      organization <- OrganizationSQLDAO.findOneByName(user.organization)
+      team <- TeamSQLDAO.findOne(_team)(GlobalAccessContext)
+      teamManagerTeamIds <- teamManagerTeamIds
+    } yield (teamManagerTeamIds.contains(_team) || this.isAdmin && this._organization == team._organization)
+
+  def isTeamManagerOrAdminOfOrg(_organization: ObjectId): Fox[Boolean] =
+    for {
+      teamManagerTeamIds <- teamManagerTeamIds
+    } yield teamManagerTeamIds.nonEmpty || this.isAdmin && this._organization == _organization
```

Comment:
`yield teamManagerTeamIds.nonEmpty || isAdminOf(_organization)`

Reply:
Also, use `isTeamManagerInOrg(_organization)` should be used, so I guess it's just
```
`isTeamManagerInOrg(_organization) || isAdminOf(_organization)
```